src/main/java/Main.java

```diff
@@ -0,0 +1,126 @@
+import spark.ModelAndView;
+import spark.Session;
+import spark.Spark;
+import spark.template.mustache.MustacheTemplateEngine;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Created by Nigel on 7/11/16.
+ */
+public class Main {
+
+    public static final String SESSION_USERNAME = "user";
+    public static final String SESSION_PASSWORD = "password";
+
+    static Map<String, User> users = new HashMap<>();
+    public static void main(String[] args) {
+        Spark.init();
+        Spark.get("/",((request, response) -> {
+
+            Session session = request.session();
+            String name = session.attribute(SESSION_USERNAME);
+            String password = session.attribute(SESSION_PASSWORD);
+
+
+            HashMap templateData = new HashMap();
+            User user = new User(name,password);
+            ModelAndView showPage = null;
+
+            if (user.getPassword() == null) {
+                showPage = new ModelAndView(templateData, "login.html");
+            }else if(users.containsKey(name)){
+                User checkUser = users.get(name);
+                if (user.getPassword().equals(checkUser.getPassword())) {
```

Comment:
This `if` does not have a corresponding `else`.  This means that you can have a situation where your `ModelAndView` is `null` which would give unexpected and bad results.